js/packages/openinference-instrumentation-openai/src/instrumentation.ts

```diff
@@ -401,15 +451,41 @@ function getChatCompletionLLMOutputMessagesAttributes(
     return {};
   }
   return [choice.message].reduce((acc, message, index) => {
-    const indexPrefix = `${SemanticConventions.LLM_OUTPUT_MESSAGES}.${index}`;
-    acc[`${indexPrefix}.${SemanticConventions.MESSAGE_CONTENT}`] = String(
-      message.content,
-    );
-    acc[`${indexPrefix}.${SemanticConventions.MESSAGE_ROLE}`] = message.role;
+    const indexPrefix = `${SemanticConventions.LLM_OUTPUT_MESSAGES}.${index}.`;
+    const messageAttributes = getChatCompletionOutputMessageAttributes(message);
+    // Flatten the attributes on the index prefix
+    for (const [key, value] of Object.entries(messageAttributes)) {
+      acc[`${indexPrefix}${key}`] = value;
+    }
     return acc;
   }, {} as Attributes);
 }
 
+function getChatCompletionOutputMessageAttributes(
+  message: ChatCompletionMessage,
+): Attributes {
+  const role = message.role;
+  const attributes: Attributes = {
+    [SemanticConventions.MESSAGE_ROLE]: role,
+  };
+  if (message.content) {
+    attributes[SemanticConventions.MESSAGE_CONTENT] = message.content;
+  }
+  if (message.tool_calls) {
+    message.tool_calls.forEach((toolCall, index) => {
+      const toolCallIndexPrefix = `${SemanticConventions.MESSAGE_TOOL_CALLS}.${index}.`;
+      attributes[
+        toolCallIndexPrefix + SemanticConventions.TOOL_CALL_FUNCTION_NAME
+      ] = toolCall.function.name;
+      attributes[
+        toolCallIndexPrefix +
+          SemanticConventions.TOOL_CALL_FUNCTION_ARGUMENTS_JSON
+      ] = JSON.stringify(toolCall.function.arguments);
+    });
+  }
+  return attributes;
```

Comment:
`function_call` is deprecated, but still works when user specifies `function=` in the request instead of `tools=`.
```suggestion
  }
  if (message.function_call) {
    ...
  }
  return attributes;
```

Reply:
Gotcha, will add